statsmodels/tsa2/sarimax/estimators/statespace.py

```diff
@@ -0,0 +1,123 @@
+        out of the likelihood. This reduces the number of parameters estimated
+        by maximum likelihood by one.
+    start_params : array_like, optional
+        Initial guess of the solution for the loglikelihood maximization. The
+        AR polynomial must be stationary. If `enforce_invertibility=True` the
+        MA poylnomial must be invertible. If not provided, default starting
+        parameters are computed using the Hannan-Rissanen method.
+    fit_kwargs : dict, optional
+        Arguments to pass to the state space model's `fit` method.
+
+    Returns
+    -------
+    parameters : SARIMAXParams object
+    other_results : Bunch
+        Includes two components, `spec`, containing the `SARIMAXSpecification`
+        instance corresponding to the input arguments; and
+        `state_space_results`, corresponding to the results from the underlying
+        state space model and Kalman filter / smoother.
+
+    Notes
+    -----
+    The primary reference is [1]_.
+
+    References
+    ----------
+    .. [1] Durbin, James, and Siem Jan Koopman. 2012.
+       Time Series Analysis by State Space Methods: Second Edition.
+       Oxford University Press.
+
+    """
+    # Handle including the constant (need to do it now so that the constant
+    # parameter can be included in the specification as part of `exog`.)
+    if include_constant:
+        exog = np.ones_like(endog) if exog is None else add_constant(exog)
+
+    # Create the specification
+    spec = SARIMAXSpecification(
+        endog, exog=exog, order=order, seasonal_order=seasonal_order,
+        enforce_stationarity=enforce_stationarity,
+        enforce_invertibility=enforce_invertibility,
+        concentrate_scale=concentrate_scale)
+    endog = spec.endog
+    exog = spec.exog
+    p = SARIMAXParams(spec=spec)
+
+    # Check start parameters
+    if start_params is not None:
+        sp = SARIMAXParams(spec=spec)
+        sp.params = start_params
+
+        if spec.enforce_stationarity and not sp.is_stationary:
+            raise ValueError('Given starting parameters imply a non-stationary'
+                             ' AR process with `enforce_stationarity=True`.')
+
+        if spec.enforce_invertibility and not sp.is_invertible:
+            raise ValueError('Given starting parameters imply a non-invertible'
+                             ' MA process with `enforce_invertibility=True`.')
+
+    # Create and fit the state space model
+    mod = SARIMAX(endog, exog=exog, order=spec.order,
+                  seasonal_order=spec.seasonal_order,
+                  enforce_stationarity=spec.enforce_stationarity,
+                  enforce_invertibility=spec.enforce_invertibility,
+                  concentrate_scale=spec.concentrate_scale)
+    if fit_kwargs is None:
+        fit_kwargs = {}
+    fit_kwargs.setdefault('disp', 0)
+    res_ss = mod.fit(start_params=start_params, **fit_kwargs)
+
+    # Construct results
+    p.params = res_ss.params
+    res = Bunch({
```

Comment:
we could consider using `types.SimpleNamespace` for these

Reply:
Should make `Bunch = types.SimpleNamespace`

Reply:
Sounds like a good idea as a general principle, but will not do it in this PR.